### PR TITLE
CLOUDP-108669: Add one AWS private endpoint for the Atlas Data Lakes in the specified project.

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-108669

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli atlas privateEndpoints dataLakes aws create --privateEndpointId vpce-0fcd9d80bbafe1607 --comment "test mongocli"
Data Lake Private endpoint 'vpce-0fcd9d80bbafe1607' created.
```

```
./bin/mongocli atlas privateEndpoints dataLakes aws create --privateEndpointId vpce-0fcd9d80bbafe1607 --comment "test mongocli" -o json
{
  "links": [
    {
      "rel": "self",
      "href": "https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/5e4e593f70dfbf1010295836/privateNetworkSettings/endpointIds?pageNum=1\u0026itemsPerPage=100"
    }
  ],
  "results": [
    {
      "comment": "test mongocli",
      "endpointId": "vpce-0fcd9d80bbafe1607",
      "provider": "AWS",
      "type": "DATA_LAKE"
    }
  ],
  "totalCount": 1
}
```

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
waiting for https://github.com/mongodb/mongocli/pull/917

**Comment by andreaangiolillo** [comment]
I will start adding e2e tests once we have the delete command.

File: internal/cli/atlas/privateendpoints/datalake/aws/create.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.privateEndpointID, flag.PrivateEndpointID, "", usage.PrivateEndpointID)
+	cmd.Flags().StringVar(&opts.comment, flag.Comment, "", usage.Comment)
+
+	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
+	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
+
+	_ = cmd.MarkFlagRequired(flag.Region)
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/datalake/aws/create.go:92]: copy/pasta?

File: internal/cli/atlas/privateendpoints/datalake/aws/create.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+				opts.initStore(cmd.Context()),
+				opts.InitOutput(cmd.OutOrStdout(), createTemplate),
+			)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.privateEndpointID, flag.PrivateEndpointID, "", usage.PrivateEndpointID)
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/datalake/aws/create.go:86]: [q] ins't this mandatory?

Comment (andreaangiolillo) [reply internal/cli/atlas/privateendpoints/datalake/aws/create.go:86]: double-checking but I believe it is

Comment (andreaangiolillo) [reply internal/cli/atlas/privateendpoints/datalake/aws/create.go:86]: yes, it is. Updating now

**Review by gssbzn — APPROVED** [review]
LGTM

File: internal/cli/atlas/privateendpoints/datalake/aws/create.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	}
+	return createRequest
+}
+
+// mongocli atlas privateEndpoint(s) dataLake aws create [--privateEndpointId privateEndpointId] [--comment comment] --projectId projectId.
+func CreateBuilder() *cobra.Command {
+	opts := &CreateOpts{}
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new Data Lake private endpoint for your project.",
```
Comment (melissamahoney-mongodb) [inline internal/cli/atlas/privateendpoints/datalake/aws/create.go:72]: Adding a little extra info here from the API docs.
```suggestion
		Short: "Create a new Data Lake private endpoint for your project.",
		Long:  `When you run this command:

- If the endpoint ID already exists and there is no change to the associated comment, Atlas Data Lake makes no change to the endpoint ID list.
- If the endpoint ID already exists and there is a change to the associated comment, Atlas Data Lake updates the comment value only in the endpoint ID list.
- If the endpoint ID doesn't exist, Atlas Data Lake appends the new endpoint to the list of endpoints in the endpoint ID list.

Your API key must have the GROUP_ATLAS_ADMIN (Project Owner) role to create a private endpoint.`
```

**Review by melissamahoney-mongodb — COMMENTED** [review]
Adding a little more info :)

**Review by gssbzn — APPROVED** [review]
LGTM

**Comment by andreaangiolillo** [comment]
I am going to merge this PR as I have addressed the Doc review comments

**Pull request merged**